api/src/main/java/org/openmrs/Condition.java

```diff
@@ -264,7 +264,7 @@ public void setOnsetDate(Date onsetDate) {
 	 * @return endDate - a date object that shows the end date of the condition
 	 */
 	public Date getEndDate() {
-		return endDate;
+		return endDate != null ? (Date) endDate.clone() : new Date();
```

Comment:
Why you are using diffrent methods here? 
1 . `new Date(onsetDate.getTime())`
2.  `(Date) endDate.clone()`
Better to keep the consistency among the code, and provide your feedback why did you select that among others.

Reply:
Thanks for pointing out that

I'm going for `(Date) endDate.clone()`, because returning `new Date(onsetDate.getTime())` is causing some tests to fail and also, using `.clone()` method makes it easy to read and understand the code.

**P.S.:** I've squashed all commits into 1

Thanks